omg/entities/player.py

```diff
@@ -51,26 +55,44 @@ def __init__(self, name, char_class, image_file, scale, initial_angle=0):
         self.mana_regen_cooldown = 0
 
         # Skills
-        self.skills = SkillManager(N_SKILLS_MAX)
+        self.element = SkillManager(N_SKILLS_MAX)
         self.item_pickup_radius = 5
+        # Initialize an empty element_buffer
+        self.to_be_combined_element_buffer: list[str] = []
+        # Initialize crafted skill slots as None
+        self.crafted_skill_slots: list[str] = [None, None]
+        self.crafted_skill = SkillFactory()
 
     def on_key_press(self, key, modifiers):
-        """Called whenever a key is pressed."""
+        """Call whenever a key is pressed."""
         self.movement_logic.on_key_press(key, modifiers)
 
-        if key == arcade.key.SPACE:
-            self.shoot()
+        if key == arcade.key.H:
+            skill_name = self.crafted_skill_slots[0]
+            self.shoot(skill_name)
+        elif key == arcade.key.J:
+            skill_name = self.crafted_skill_slots[1]
+            self.shoot(skill_name)
+        elif key == arcade.key.SPACE:
+            self.to_be_combined_element_buffer.append(
+                self.element.get_current()["name"]
```

Comment:
This line creates a bug, Pressing Space with no element in self.element causes a TypeError. You can replicate by just launching the game and press Space